torch/nn/quantized/modules/conv.py

```diff
@@ -54,6 +54,8 @@ def _init(self, in_channels, out_channels, kernel_size, stride,
         self.out_channels = out_channels
         self.kernel_size = kernel_size
         self.stride = stride
+        if isinstance(padding, str) or (isinstance(padding, tuple) and isinstance(padding[0], str)):
+            raise ValueError('padding must be a numerical value')
```

Comment:
I think the padding is supposed to be int, tuple, or str? At least according to the nn.ConvXd docs: https://pytorch.org/docs/stable/generated/torch.nn.Conv1d.html

Reply:
The problem is the quantized kernel doesn't actually handle them, it only accepts a list[int] which is what a tuple ad this point gets converted to. See the github issue this PR is in reference to. note: We're looking for tuple(str) because for conv 2d or whatever, when they do _single or _pair on the string, that's the result.

Reply:
I don't have a super-strong opinion about when the string arguments should be implemented, but I do believe that quant API should be the same as the FP. Ipso facto, I would rather see the padding handling be fixed, rather than telling the users that we don't support that.

---

> The problem is the quantized kernel doesn't actually handle them,

The kernels don't really have to make sure the arguments are not malformed. On the python side, we can just add all the necessary plumbing, without being blocked by the kernel dev.

With that said, I agree that the FP kernels do have support for string padding. In FP the conv kernels are overloaded, and there are two implementations:
1. [with type of padding being `IntArrayRef`](https://github.com/pytorch/pytorch/blob/ba818991afb4ef6189fa4ed57dd204173e9c7b74/aten/src/ATen/native/Convolution.cpp#L764-L781) -- this is just taking the padding as an array of integers.
2. [with type of padding being `std::string`](https://github.com/pytorch/pytorch/blob/ba818991afb4ef6189fa4ed57dd204173e9c7b74/aten/src/ATen/native/Convolution.cpp#L906-L920) -- this does the [same conversion](https://github.com/pytorch/pytorch/blob/ba818991afb4ef6189fa4ed57dd204173e9c7b74/aten/src/ATen/native/Convolution.cpp#L847-L857) as [in python](https://github.com/pytorch/pytorch/blob/ba818991afb4ef6189fa4ed57dd204173e9c7b74/torch/nn/modules/conv.py#L114-L123), and calls the convolution described in (1)

I think the reason there are two versions of the convXd is because one can call `F.conv1d(padding='same')`, and it is necessary by API to support that. It is not super easy to implement the conversion directly in the `nn/functional.py`, as it [uses an `_add_docstr` wrapper](https://github.com/pytorch/pytorch/blob/ba818991afb4ef6189fa4ed57dd204173e9c7b74/torch/nn/functional.py#L30), instead of direct function definition.

However, for the quantized we don't have to do it that way:

1. In the `nn.quantized.ConvXd` we can add a routine to convert the padding
    1. 'valid': `self.padding=0`
    2. 'same': `self.padding=[self.kernel_size//2, self.kernel_size//2]` (in practice, we need to handle the dilation and different dimensionality).
1. In the [`nn.quantized.functional.convXd`](https://github.com/pytorch/pytorch/blob/ba818991afb4ef6189fa4ed57dd204173e9c7b74/torch/nn/quantized/functional.py#L111-L168) we can add the same routine to convert the padding